src/nomad_simulations/properties/spectral_profile.py

```diff
@@ -0,0 +1,601 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+from structlog.stdlib import BoundLogger
+from typing import Optional, List, Dict
+import pint
+
+from nomad import config
+from nomad.metainfo import Quantity, SubSection, Section, Context
+
+from ..utils import get_sibling_section
+from ..physical_property import PhysicalProperty
+from ..variables import Energy2 as Energy
+from ..atoms_state import AtomsState
+from .band_gap import ElectronicBandGap
+
+
+class SpectralProfile(PhysicalProperty):
+    """
+    A base section used to define the spectral profile.
+    """
+
+    value = Quantity(
+        type=np.float64,
+        description="""
+        The value of the intensities of a spectral profile in arbitrary units.
+        """,
+    )
+
+    # TODO implement this in PhysicalProperty (similar to `value.shape`)
+    # value_units = Quantity(
+    #     type=str,
+    #     description="""
+    #     Unit using the pint UnitRegistry() notation for the `value`. Example, if the spectra `is_derived`
+    #     from the imaginary part of the dielectric function, `value` are `'F/m'`.
+    #     """,
+    # )
+
+    def __init__(
+        self, m_def: Section = None, m_context: Context = None, **kwargs
+    ) -> None:
+        super().__init__(m_def, m_context, **kwargs)
+        self.rank = []  # ? Is this here or in the attrs instantiation better?
+
+    def is_valid_spectral_profile(self) -> bool:
+        """
+        Check if the spectral profile is valid, i.e., if all `value` are defined positive.
+
+        Returns:
+            (bool): True if the spectral profile is valid, False otherwise.
+        """
+        if (self.value < 0.0).any():
+            return False
+        return True
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+        if self.is_valid_spectral_profile() is False:
+            logger.error(
+                'Invalid negative intensities found: could not validate spectral profile.'
+            )
+            return
+
+
+class DOSProfile(SpectralProfile):
+    """
+    A base section used to define the `value` of the `ElectronicDensityOfState` property. This is useful when containing
+    contributions for `projected_dos` with the correct unit.
+    """
+
+    value = Quantity(
+        type=np.float64,
+        unit='1/joule',
+        description="""
+        The value of the electronic DOS.
+        """,
+    )
+
+    def __init__(
+        self, m_def: Section = None, m_context: Context = None, **kwargs
+    ) -> None:
+        super().__init__(m_def, m_context, **kwargs)
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+        # TODO add normalization for `projected_dos` to extract `name` (if `m_parent` is `ElectronicDOS`
+        # TODO (cont'd) and `entity_ref` is set)
```

Comment:
I need this class indeed to have `value` set up with the proper `'1/joule'` units in `projected_dos`. Otherwise, I need to point to `SpectralProfile` and there `value` does not have units (hence `projected_dos.value` does not either).

It is a bit complicated but it is the only way of keeping these contributions with the proper units in their values.

Reply:
I see. Still, the DOS does not require any special treatment. You should rather rethink how you handle spectra in general: in the end, all spectra lie on an energy or frequency axis (with a simple inter-conversion), right? Their amplitudes are completely defined by that axis and what they integrate to. Idk if there's a generic dimensionality for the latter, maybe _quasi-particles_?

Most scientists don't bother labelling the amplitude axis, since they only care for relative heights inside a single sample. Likewise, most DOS is also reported in a.u. This approach just prevents _inter-sample comparison_, which might be wishful.

_Implementation suggestion_
If `Quantity` cannot accommodate your needs, go with a dynamic check, as you've done before. Or you forego the check, if you can't formulate one.

Reply:
It is not like that: `projected_dos` is a list of elements which can contain atom-pdos or orbital-pdos, something like:
```python
projected_dos = [
   atom Ga:DOSProfile(name='atom Ga', value=...), 
   orbital px Ga:DOSProfile(name='orbital px Ga', value=...), 
   orbital py Ga:DOSProfile(name='orbital py Ga', value=...),
]
```

Hence the special requirement of looping and getting one name or another matched (see `extract_projected_dos(type, logger)`)

Reply:
But this is a different consideration than the one you mentioned before regarding the units:
> I need this class indeed to have value set up with the proper '1/joule' units in projected_dos

All I'm saying is that you could set generic units (e.g. `qp_mass/joule`) for spectra, no need for `DOSProfile`. You can determine the amplitude units from the energy axis (converting frequency axes) and the no. QPs.

I guess all spectra should be able to contain contributions of other spectra.
These can then either be used to construct those, as with the pDOS, or concatenated, as with XAS and XANES.
These specific functionalities (interfaces) could either both be specified at `Spectra` or (maybe better) in the child classes representing the 3 techniques.